js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+                'private': { 'post': [''] },
+            },
+            'fees': {
+                'trading': {
+                    'tierBased': false,
+                    'percentage': true,
+                    'maker': 0,
+                    'taker': 0.3,
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const tickerRes = await this.publicGet ({ 'method': 'getticker' });
+        const tickersData = this.parseJson (tickerRes);
+        const coinRes = await this.publicGet ({ 'method': 'getcoins' });
+        const coinResponse = this.parseJson (coinRes);
+        const tickerIds = Object.keys (tickersData);
+        let result = [];
+        for (let i = 0; i < tickerIds.length; i++) {
+            const id = tickerIds[i];
+            const tickerData = tickersData[id];
+            const splitId = id.split ('_');
+            const baseId = splitId[0]; // base
+            const quoteId = splitId[1]; // quote
+            const base = this.commonCurrencyCode (baseId);
+            const quote = this.commonCurrencyCode (quoteId);
+            const active = tickerData.isFrozen === 0;
+            const coinData = coinResponse[quoteId];
+            const market = {
+                'id': id,
+                'symbol': base + '/' + quote,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': active,
+                'maker': coinData['makerfee'],
+                'taker': coinData['takerfee'],
+                // precision: precision, // TODO: not listed? infer?
+                'info': tickerData,
+                // limits: limits,
+            };
+            result.push (market);
+        }
+        return result;
+    }
+
+    async fetchCurrencies (params = {}) {
+        const res = await this.publicGet ({ 'method': 'getcoins' });
+        const currenciesData = this.parseJson (res);
+        const currencyIds = Object.keys (currenciesData);
+        // The API does not expose BTC as a coin (I suspect because it is the base in each market)
+        let result = { 'BTC': {
+            'id': 'BTC',
+            'code': 'BTC',
+            'name': 'bitcoin',
+            'fiat': false,
+        }};
+        for (let i = 0; i < currencyIds.length; i++) {
+            const id = currencyIds[i];
+            const coinData = currenciesData[id];
+            const code = this.commonCurrencyCode (id);
+            result[code] = {
+                'id': id,
+                'code': code,
+                'name': coinData['name'],
+                'fiat': false,
+                'funding': {
+                    'withdraw': {
+                        'fee': this.safeFloat (coinData, 'withdrawfee'),
+                    },
+                },
+            };
+        }
+        return result;
+    }
+
+    fetchTradingFees (params = {}) {
```

Comment:
I would leave out this function.  It's usually used if there's some sort of a tiered system, and gives you personal info about your own trading fees.  For example, if you trade more than 1 million in a month, some exchange fees can go down from 0.5% to 0.2%, and I'm still on the first tier, fetchTradingFees would give you 0.2% and me 0.5%

Reply:
sure--also they're available on the fetchMarket call and actually come from the API. Removed.